source/patterns/@aws-solutions-constructs/aws-openapigateway-lambda/lib/openapi-helper.ts

```diff
@@ -0,0 +1,171 @@
+    errorFound = true;
+  }
+
+  if (props.apiIntegrations === undefined || props.apiIntegrations.length < 1) {
+    errorMessages += 'At least one ApiIntegration must be specified in the apiIntegrations property\n';
+    errorFound = true;
+  }
+
+  if (errorFound) {
+    throw new Error(errorMessages);
+  }
+
+}
+
+interface ObtainApiDefinitionProps {
+  readonly apiLambdaFunctions: ApiLambdaFunction[],
+  readonly apiDefinitionBucket?: s3.IBucket,
+  readonly apiDefinitionKey?: string,
+  readonly apiDefinitionAsset?: Asset,
+  readonly apiJsonDefinition?: any,
+  readonly internalTransformTimeout?: Duration,
+  readonly internalTransformMemorySize?: number
+}
+
+export function ObtainApiDefinition(scope: Construct, props: ObtainApiDefinitionProps): apigateway.ApiDefinition {
+  const apiRawInlineSpec = props.apiJsonDefinition;
+  const apiDefinitionBucket = props.apiDefinitionBucket ?? props.apiDefinitionAsset?.bucket;
+  const apiDefinitionKey = props.apiDefinitionKey ?? props.apiDefinitionAsset?.s3ObjectKey;
+
+  // Map each id and lambda function pair to the required format for the template writer custom resource
+  const apiIntegrationUris = props.apiLambdaFunctions.map(apiLambdaFunction => {
+    // the placeholder string that will be replaced in the OpenAPI Definition
+    const uriPlaceholderString = apiLambdaFunction.id;
+    // the endpoint URI of the backing lambda function, as defined in the API Gateway extensions for OpenAPI here:
+    // https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-integration.html
+    const uriResolvedValue = `arn:${Aws.PARTITION}:apigateway:${Aws.REGION}:lambda:path/2015-03-31/functions/${apiLambdaFunction.lambdaFunction.functionArn}/invocations`;
+
+    return {
+      id: uriPlaceholderString,
+      value: uriResolvedValue
+    };
+  });
+
+  let apiDefinitionWriter: resources.CreateTemplateWriterResponse | undefined;
+  let newApiDefinition: apigateway.ApiDefinition | undefined;
+
+  if (props.apiDefinitionAsset || props.apiDefinitionBucket) {
+    // This custom resource will overwrite the string placeholders in the openapi definition with the resolved values of the lambda URIs
+    apiDefinitionWriter = resources.createTemplateWriterCustomResource(scope, 'Api', {
+      // CheckOpenapiProps() has confirmed the existence of these values
+      templateBucket: apiDefinitionBucket!,
+      templateKey: apiDefinitionKey!,
+      templateValues: apiIntegrationUris,
+      timeout: props.internalTransformTimeout ?? Duration.minutes(1),
+      memorySize: props.internalTransformMemorySize ?? 1024
+    });
+
+    newApiDefinition = apigateway.ApiDefinition.fromBucket(
+      apiDefinitionWriter.s3Bucket,
+      apiDefinitionWriter.s3Key
+    );
+  } else if (apiRawInlineSpec) {
+    const apiInlineSpec = new apigateway.InlineApiDefinition(apiRawInlineSpec);
```

Comment:
Perhaps this isn't needed since `apiRawInlineSpec` is technically already expected to be a JSON object. Unless you are relying on the initial assignment to trigger validation, which would make sense in this instance. Non-and empty objects would throw.

Reply:
Yep, code is redundant to the check already done in CheckOpenapiProps. But including it makes the code explicit, these steps aren't just executing because it is "not something else", it is executing because it IS "this". Putting that in code is better than a comment - even at the cost of a couple clock ticks.